### PR TITLE
Mrmc backports

### PR DESCRIPTION
**Pull request by Memphiz**
As stated in the forum - some backports from the MrMC fork. Anyone here to comment? I tried to only pick the sane ones ...

### CODE REVIEW

**Comment by Montellese** [comment]
Partly overlaps with #7986

**Comment by stefansaraev** [comment]
https://github.com/stefansaraev/xbmc/commit/b3f4d14f4c09883c2eb64446916d48d6f83f3bdf https://github.com/stefansaraev/xbmc/commit/8b8c7d548384e3174fe935cb3896a561a20b345b and I'll close #7986

**Comment by Memphiz** [comment]
oh sorry - i still am not used to not getting all the github spam *grr*. (so i clearly miss things).

I have picked those @stefansaraev - thx

**Comment by MrMC** [comment]
Beware the SDL removal. seems ok now but mouse/keyboard handling needs more work. Sometimes the mouse is lost on start and keyboard is not completely hooked

**Comment by MrMC** [comment]
Ah, nm. that's native, not OSX target :)

**Comment by MrMC** [comment]
Ah, nm. that's native, not OSX target :)

**Comment by Memphiz** [comment]
I leave that osx target sdl removal cooking a bit more on your side *cough* before i bring it to the table ... ;)

**Comment by stefansaraev** [comment]
jenkins build this please :)

**Comment by Memphiz** [comment]
all fine but windows - fixed the vsproj. file and kicked a win only build here:

http://jenkins.kodi.tv/job/WIN-32/6088/

**Comment by Memphiz** [comment]
builds fine - any objections? (or an ok from someone else) - the next wave of fixes is rolling on already - don't want to do it in one big PR but multiple smaller ones...

File: xbmc/dialogs/GUIDialogKeyboardGeneric.cpp [hunk 1]
```diff
@@ -600,7 +600,7 @@ void CGUIDialogKeyboardGeneric::ChangeWordList(int direct)
   else
   {
     ShowWordList(direct);
-    if (direct > 0 && m_pos + m_num == m_words.size())
+    if (direct > 0 && m_pos + m_num == (int)m_words.size())
```
Comment (Paxxi) [inline xbmc/dialogs/GUIDialogKeyboardGeneric.cpp]: Use proper c++ style casts. Might be annoying to update but would be nice

**Comment by Paxxi** [comment]
Looks good, would be nice if the nitpicking is fixed but I won't block the pr based on that

File: xbmc/dialogs/GUIDialogKeyboardGeneric.cpp (hunk 1 above)
Comment (Memphiz) [inline xbmc/dialogs/GUIDialogKeyboardGeneric.cpp]: all static_casts right?

Comment (fritsch) [inline xbmc/dialogs/GUIDialogKeyboardGeneric.cpp]: Yes.

2015-09-10 12:15 GMT+02:00 Memphiz <notifications@github.com>:

> In xbmc/dialogs/GUIDialogKeyboardGeneric.cpp
> <https://github.com/xbmc/xbmc/pull/7991#discussion_r39144253>:
>
> > @@ -600,7 +600,7 @@ void CGUIDialogKeyboardGeneric::ChangeWordList(int direct)
> >    else
> >    {
> >      ShowWordList(direct);
> > -    if (direct > 0 && m_pos + m_num == m_words.size())
> > +    if (direct > 0 && m_pos + m_num == (int)m_words.size())
>
> all static_casts right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/7991/files#r39144253>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

**Comment by Memphiz** [comment]
done - started a testbuild on osx to ensure i didn't fatfinger it (have no compiler at hand atm).

**Comment by fritsch** [comment]
Btw.: http://www.fatfingers.co.uk/

**Comment by MartijnKaijser** [comment]
seems to have build fine

**Comment by MartijnKaijser** [comment]
@Memphiz you created that revert branch by accident?

**Comment by Memphiz** [comment]
yep ... wanted to remove my branch when that button wasn't shown yet ... could you remove it @MartijnKaijser - i don't see how to do it from the github webfrontend

**Comment by Montellese** [comment]
Done

**Comment by Memphiz** [comment]
thx :)

**Pull request merged**